docs/docs/tutorials/simple_hello_world.md

```diff
@@ -0,0 +1,198 @@
+--go_out=Mgoogle/api/annotations.proto=github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis/google/api,plugins=grpc:. \
+helloworld/helloworld.proto
+```
+
+The `helloworld.pb.go` file will be generated. So `helloworld.pb.go` is required by the server service.
+
+Next we need to use protoc to generate the go files needed by the gateway.
+
+```sh
+protoc -I/usr/local/include -I. \
+-I$GOPATH/src  -I$GOPATH/src/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis \
+--swagger_out=logtostderr=true:. \
+helloworld/helloworld.proto
+```
+
+The `helloworld.pb.gw.go` file will be generated. This file is the protocol file used by gateway for protocol conversion between grpc and http.
+
+After the protocol file is processed, the gateway code is needed.
+
+### Implementation
+
+The gateway code is as follows:
+
+```go
+package main
+
+import (
+    "flag"
+    "net/http"
+
+    "github.com/golang/glog"
+    "github.com/grpc-ecosystem/grpc-gateway/runtime"
+    "golang.org/x/net/context"
+    "google.golang.org/grpc"
+
+    gw "grpc-helloworld-gateway/helloworld"
+)
+
+var (
+    echoEndpoint = flag.String("echo_endpoint", "localhost:50051", "endpoint of YourService")
+)
+
+func run() error {
+    ctx := context.Background()
+    ctx, cancel := context.WithCancel(ctx)
+    defer cancel()
+
+    mux := runtime.NewServeMux()
+    opts := []grpc.DialOption{grpc.WithInsecure()}
+    err := gw.RegisterGreeterHandlerFromEndpoint(ctx, mux, *echoEndpoint, opts)
+    if err != nil {
+        return err
+    }
+
+    return http.ListenAndServe(":8080", mux)
+}
+
+func main() {
+    flag.Parse()
+    defer glog.Flush()
+
+    if err := run(); err != nil {
+        glog.Fatal(err)
+    }
+}
+```
+
+First echoEndpoint stores the server information that needs to be connected, and then registers and binds the new server with RegisterGreeter Handler FromEndpoint in `gw.go`. Then the lower level will connect to the remote server address provided by echoEndpoint, so the gateway establishes the connection as the client and remote server, and then starts the new server with http, gateway. It serves as a server to provide HTTP services to the outside world.
+
+This is the end of the code. Let's test it.
+
+Start the greeter_server service first, and then start the gateway. Then gateway connects to greeter_server and establishes http monitoring.
+
+Then we use curl to send http requests:
+
+```sh
+curl -X POST -k http://localhost:8080/v1/example/echo -d '{"name": " world"}
+```
+
+```
+{"message":"Hello  world"}
+```
+
+The process is as follows:
+
+curl sends a request to the gateway with the post, gateway as proxy forwards the request to greeter_server through grpc, greeter_server returns the result through grpc, the gateway receives the result, and json returns to the front end.
+
+In this way, the transformation process from http json to internal grpc is completed through grpc-gateway.
```

Comment:
This is great, definitely keep this, but move it to a new page that talks about adding the grpc-gateway annotations to an existing protobuf file.